docs-src/docs/guides/terraform_beyond_layer0.md

```diff
@@ -150,7 +150,180 @@ This is then used to populate the template fields in our [Dockerrun.aws.json](ht
+  load_balancer = "${layer0_load_balancer.guestbook.id}"
+  scale         = 3
+}
+
+```
+
+### Plan and Apply
+Once you have updated the following command to understand the changes that you will be making. Note that if you did not specify `scale` it defaults to '1'.
+
+`terraform plan`
+
+Outputs:
+
+```
+Refreshing Terraform state in-memory prior to plan...
+The refreshed state will be used to calculate this plan, but will not be
+persisted to local or remote state storage.
+
+layer0_environment.demo: Refreshing state... (ID: demoenvbb9f6)
+data.template_file.guestbook: Refreshing state...
+layer0_deploy.guestbook: Refreshing state... (ID: guestbook.6)
+layer0_load_balancer.guestbook: Refreshing state... (ID: guestbo43ab0)
+layer0_service.guestbook: Refreshing state... (ID: guestboebca1)
+The Terraform execution plan has been generated and is shown below.
+Resources are shown in alphabetical order for quick scanning. Green resources
+will be created (or destroyed and then created if an existing resource
+exists), yellow resources are being changed in-place, and red resources
+will be destroyed. Cyan entries are data sources to be read.
+
+Note: You didn't specify an "-out" parameter to save this plan, so when
+"apply" is called, Terraform can't guarantee this is what will execute.
+
+~ layer0_service.guestbook
+    scale: "1" => "3"
+```
+
+Now run the following command to deploy your changes:
+
+`terraform apply`
+
+Outputs:
+
+```
+layer0_environment.demo: Refreshing state... (ID: demoenvbb9f6)
+data.template_file.guestbook: Refreshing state...
+layer0_deploy.guestbook: Refreshing state... (ID: guestbook.6)
+layer0_load_balancer.guestbook: Refreshing state... (ID: guestbo43ab0)
+layer0_service.guestbook: Refreshing state... (ID: guestboebca1)
+layer0_service.guestbook: Modifying... (ID: guestboebca1)
+  scale: "1" => "3"
+layer0_service.guestbook: Modifications complete (ID: guestboebca1)
+
+Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
+
+The state of your infrastructure has been saved to the path
+below. This state is required to modify and destroy your
+infrastructure, so keep it safe. To inspect the complete state
+use the `terraform show` command.
+
+State path: 
+
+Outputs:
+
+guestbook_url = <guestbook_service_url>
+```
+
+To confirm your service has scaled, you can run the following layer0 command. Note desired scale for the guestbook service should be eventually be 3/3.
+
+`l0 service get \*`
+
+Outputs:
+
+```
+SERVICE ID    SERVICE NAME  ENVIRONMENT  LOADBALANCER  DEPLOYMENTS  SCALE
+api           api           api          api           api:3        1/1
+guestboebca1  guestbook     demo-env     guestbook     guestbook:6  3/3
+```
+
+
+
+## Part 6: Terraform Destroy
 When you're finished with the example run the following command in the same directory to destroy the Layer0 environment, application and the DynamoDB Table.
 
 `terraform destroy`
+
+## Best Practices with Terraform + Layer0
```

Comment:
basically i was thinking this section would cover "devops" type questions:

* how do i manage tfstate? (remote backend)
* how do i manage sharing variables? (best practice - don't put it in you source control!), https://www.vaultproject.io/ could potentially help?
* always `terraform plan` before `terraform apply`
* we don't have tasks in terraform 
* How do i organize my terraform configurations? What if i have two services in the same environment?
Answer: use modules https://www.terraform.io/docs/modules/usage.html (example https://github.com/quintilesims/consul/tree/master/terraform). The high level answer is "configure your terraform files to be consumable as a module". 
* Make docker image and/or tag configurable. Don't always use `latest`. 
* Good source of modules: https://github.com/terraform-community-modules